src/main/java/seedu/address/model/person/PersonDetailContainsKeywordAndExamPredicate.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.model.person;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LESSTHAN;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MORETHAN;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.model.exam.Exam;
+
+/**
+ * Tests that a {@code Person}'s details contains the keyword given.
+ */
+public class PersonDetailContainsKeywordAndExamPredicate implements Predicate<Person> {
+    private final Prefix prefix;
+    private final String keyword;
+    private final Exam exam;
+
+    /**
+     * Constructor for the PersonDetailContainsKeywordPredicate.
+     * @param prefix The prefix to indicate the detail of the person to be searched.
+     * @param keyword The keyword to be searched.
+     */
+    public PersonDetailContainsKeywordAndExamPredicate(Prefix prefix, String keyword, Exam exam) {
+        this.prefix = prefix;
+        this.keyword = keyword;
+        this.exam = exam;
+    }
+
+    /**
+     * Tests if the person's details contain the keyword.
+     */
+    @Override
+    public boolean test(Person person) {
```

Comment:
is there a way to remove the nested if-else?

Reply:
I could abstract this further into helper classes, thanks for pointing this out!